packages/guides/src/Compiler/NodeTransformers/FootNodeNamedTransformer.php

```diff
@@ -19,11 +20,15 @@ public function __construct(
     ) {
     }
 
-    public function enterNode(Node $node, DocumentNode $documentNode): Node
+    public function enterNode(Node $node, CompilerContext $compilerContext): Node
     {
+        if ($compilerContext->getDocumentNode() === null) {
```

Comment:
I think it is good to move this test to the `getDocumentNode` method, to reduce the duplication of code.